File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaRMIServidor/src/pe/edu/pucp/softbiblioteca/rmi/interfaces/ReservaBO.java

package pe.edu.pucp.softbiblioteca.rmi.interfaces;

import java.util.ArrayList;
import pe.edu.pucp.softbiblioteca.reservas.model.Reserva;
import java.rmi.Remote;
import java.rmi.RemoteException;

public interface ReservaBO extends Remote{
    
    public Integer insertar(String instanteReserva,String instanteRecojo,String tipoEstado, 
                            Integer idEjemplarAsociado,Integer idUsuarioAsociado) throws RemoteException;
    
    public Integer modificar(Integer idReserva,String instanteReserva,String instanteRecojo,String tipoEstado, 
                            Integer idEjemplarAsociado,Integer idUsuarioAsociado) throws RemoteException;
    
    public Integer eliminar(Integer idReserva) throws RemoteException;
    
    public ArrayList<Reserva> listarTodos(String instanteReserva,String instanteRecojo,
                                          String tipoEstado,Integer idEjemplarAsociado,
                                          Integer idUsuarioAsociado, Integer limite) throws RemoteException;
    
    public Reserva obtenerPorId(Integer idReserva) throws RemoteException;
    
    public ArrayList<Reserva> mostrarReservasEnCurso(Integer idUsuario) throws RemoteException;
    
    public Integer cancelarReserva(Integer idReserva) throws RemoteException;

    public Reserva solicitarReserva(Integer idUsuario, Integer idPublicacion,String tipoPublicacion) throws RemoteException;
    
    public Integer confirmarReserva(Integer idUsuario, Integer idEjemplarAsociado, String instanteRecojoMax) throws RemoteException;
    
}

File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaReservasBO/ReservaBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;

namespace SoftBibliotecaUsuarioBO
{
    public class ReservaBO : BaseBO
    {

        /*
        public int insertar(string instanteReserva, string instanteRecojo, string tipoEstado,
                                int idEjemplarAsociado, int idUsuarioAsociado)
        {
            return this.WsCliente.reserva_insertar(instanteReserva, instanteRecojo, tipoEstado,idEjemplarAsociado,idUsuarioAsociado);
        }

        public int modificar(int idReserva,string instanteReserva, string instanteRecojo, string tipoEstado,
                                int idEjemplarAsociado, int idUsuarioAsociado)
        {
            return this.WsCliente.reserva_modificar(idReserva,instanteReserva, instanteRecojo, tipoEstado, idEjemplarAsociado, idUsuarioAsociado);
        }

        public int eliminar(int idReserva)
        {
            return this.WsCliente.reserva_eliminar(idReserva);
        }

        public BindingList<reserva> listarTodos()
        {
            try
            {
                reserva[] reservas = this.WsCliente.reserva_listarTodos();
                if (reservas == null) return new BindingList<reserva>();
                return new BindingList<reserva>(reservas);
            }
            catch (Exception e)
            {
                return new BindingList<reserva>();
            }
        }

        public BindingList<reserva> listarTodos(string instanteReserva, string instanteRecojo,
                                                string tipoEstado, int idEjemplarAsociado, int idUsuarioAsociado)
        {
            try
            {
                reserva[] reservas = this.WsCliente.reserva_listarTodos_porParametros(instanteReserva, instanteRecojo, tipoEstado, idEjemplarAsociado, idUsuarioAsociado);
                if (reservas == null) return new BindingList<reserva>();
                return new BindingList<reserva>(reservas);
            }
            catch (Exception e)
            {
                return new BindingList<reserva>();
            }
        }

        public reserva obtenerPorId(int idReserva)
        {
            return this.WsCliente.reserva_obtenerPorId(idReserva);
        }
        */


        public BindingList<reserva> mostrarReservasEnCurso(int idUsuario)
        {
            try
            {
                reserva[] reservas = this.WsCliente.mostrarReservasEnCurso(idUsuario);
                if (reservas == null) return new BindingList<reserva>();
                return new BindingList<reserva>(reservas);
            }
            catch (Exception e)
            {
                return new BindingList<reserva>();
            }
        }

        public int cancelarReserva(int idReserva)
        {
            return this.WsCliente.cancelarReserva(idReserva);
        }
        //cambios giano reservaBO
        public reserva solicitarReserva(int idUsuario, int idPublicacion, string tipoPublicacion)
        {
            return this.WsCliente.solicitarReserva(idUsuario, idPublicacion, tipoPublicacion);

        }
        public int confirmarReserva(int idUsuario, int idEjemplarAsociado, string instanteRecojoMax)
        {
            return this.WsCliente.confirmarReserva(idUsuario, idEjemplarAsociado, instanteRecojoMax);

        }
        //cambios fin
    }
}
